spmd/tensor/api.py

```diff
@@ -74,12 +86,10 @@ def __torch_function__(cls, func, types, args=(), kwargs={}):
         # if we find nn.functional name in dispatch op, dispatch to it instead,
         # this allow us to override some python level behaviors that wouldn't be
         # possible in __torch_dispatch__ level.
-        if func.__name__ in DTensor._dist_tensor_dispatch_ops:
+        if func.__name__ in DTensor._custom_dispatch_ops:
```

Comment:
Could it happen that there's a torch.function with the same name as a torch dispatch function? IN that case which would take priority? also is func.__name__ the same as str(func) (in the dispatch below?)

Reply:
torch_function and torch_dispatch func.__name__ should be different for every operator, bc torch_dispatch will append a "default" to every op while torch_function won't have it. Example difference for op `mm`:

torch function `func.__name__`: `mm`
torch function `str(func)`: `<built-in method mm of type object at 0x10a5eb380>`
torch dispatch `func.__name__`: `mm.default`
torch dispatch `str(func)`: `aten.mm.default`

we are using the first and last one. I think the last one should be proper to distinguish clearly with torch_function, where it always work on `aten` ops